app/src/main/java/session/SharedPreferenceRepository.java

```diff
@@ -16,24 +16,25 @@
 import domain.Coupon;
 import domain.Offer;
 
+/**
+ * Used for saving coupons into memory on the mobile device. This will be replaced by a proper
+ * data storage system.
+ */
 public class SharedPreferenceRepository implements ISessionRepository {
     private final Gson gson;
-    private Context contextOfApplication;
     private SharedPreferences prefs;
 
-
     public SharedPreferenceRepository(Context contextOfApplication) {
-        this.contextOfApplication = contextOfApplication;
         this.prefs = PreferenceManager.getDefaultSharedPreferences(contextOfApplication);
         this.gson = new Gson();
     }
 
-    private List<Coupon> GetUserCouponsInPref(String userId){
+    private List<Coupon> getUserCouponsInPref(String userId){
         Type couponListType = new TypeToken<ArrayList<Coupon>>(){}.getType();
 
         String couponsJson = prefs.getString(userId, "");
 
-        if(couponsJson == null || couponsJson.isEmpty())
+        if(couponsJson.isEmpty())
```

Comment:
If prefs.getString(userId, "") returns null, because the userId is wrong or doesn't exist, then couponsJson becomes null and if you call couponsJson.isEmpty it will give a null pointer exception